cql3/expr/to_restriction.cc

```diff
@@ -67,10 +67,7 @@ void validate_single_column_relation(const column_value& lhs, oper_t oper, const
     }
 
     if (is_lhs_subscripted) {
-        check_false(lhs_col_type.is_list(),
-                    "Indexes on list entries ({}[index] = value) are not currently supported.",
-                    lhs_col_name);
-        check_true(lhs_col_type.is_map(), "Column {} cannot be used as a map", lhs_col_name);
+        check_true(lhs_col_type.is_map() || lhs_col_type.is_list(), "Column {} cannot be subscripted", lhs_col_name);
         check_true(lhs_col_type.is_multi_cell(),
                    "Map-entry equality predicates on frozen map column {} are not supported",
```

Comment:
It would be good to change other error messages to include lists alongside maps.
Right now if someone attempts to make an indexed restriction on non-frozen list they will get a message about maps.

Reply:
Adjusted. I think those restrictions are wrong btw. When filtering we can easily support non-equality relations or frozen collections. This check belongs in query planning when trying to select the index, not in validation.

Reply:
E.g., https://github.com/scylladb/scylla/issues/9777